Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut

```diff
@@ -1306,14 +1312,138 @@ void function RemoveConnectToServerCallback( void functionref( ServerInfo ) call
-	{
+	if ( !targetServer )
 		targetServer = file.lastSelectedServer
-	}
+
+	expect ServerInfo( targetServer )
 
 	foreach( callback in file.connectCallbacks )
 	{
-		callback( expect ServerInfo( targetServer ) )
+		callback( targetServer )
+	}
+}
+
+//////////////////////////////////////
+// Join server
+//////////////////////////////////////
+
+void function CancelJoinServer()
+{
+	file.cancelConnection = true
+}
+
+bool function JoinServerByName( string serverName, string password = "" )
+{
+	// Request list of online servers.
+	NSRequestServerList()
+	while ( NSIsRequestingServerList() )
+	{
+		WaitFrame() 
+	}
+
+	// Go through all servers that are currently online
+	foreach ( server in NSGetGameServers() )
+	{
+		// Join the server if it has the correct server name.
+		if ( server.name == serverName )
+		{
+			return JoinServer( server )
+		}
 	}
+
+	print( format( "Failed to connect to server %s: No such server", serverName ) )
+
+	return false
+}
+
+bool function JoinServer( ServerInfo server, string password = "" )
+{
+	if ( NSIsAuthenticatingWithServer() )
+		return false
+
+	printt( format( "Connecting to server %s with id of %s", server.name, server.id ) )
+
+	TriggerConnectToServerCallbacks( server )
+	NSTryAuthWithServer( server.index, password )
+
+	ToggleConnectingHUD( true )
+
+	while ( NSIsAuthenticatingWithServer() && !file.cancelConnection )
+		WaitFrame()
+
+	ToggleConnectingHUD( false )
+
+	if ( file.cancelConnection )
+	{
+		file.cancelConnection = false
+		return false
+	}
+
+	file.cancelConnection = false
+
+	if ( NSWasAuthSuccessful() )
+	{
+		bool modsChanged = false
+
+		// disable all RequiredOnClient mods that are not required by the server and are currently enabled
+		foreach ( string modName in NSGetModNames() )
+		{
+			if ( NSIsModRequiredOnClient( modName ) && NSIsModEnabled( modName ) )
+			{
+				// find the mod name in the list of server required mods
+				bool found = false
+				foreach ( RequiredModInfo mod in server.requiredMods )
+				{
+					if ( mod.name == modName )
+					{
+						found = true
+						break
+					}
+				}
+				// if we didnt find the mod name, disable the mod
+				if ( !found )
+				{
+					modsChanged = true
+					NSSetModEnabled( modName, false )
+				}
+			}
+		}
+
+		// enable all RequiredOnClient mods that are required by the server and are currently disabled
+		foreach ( RequiredModInfo mod in server.requiredMods )
+		{
+			if ( NSIsModRequiredOnClient( mod.name ) && !NSIsModEnabled( mod.name ) )
+			{
+				modsChanged = true
+				NSSetModEnabled( mod.name, true )
+			}
+		}
+
+		// only actually reload if we need to since the uiscript reset on reload lags hard
+		if ( modsChanged )
+			ReloadMods()
+
+		NSConnectToAuthedServer()
+		return true
+	}
+	else
+	{
+		string reason = NSGetAuthFailReason()
+
+		DialogData dialogData
+		dialogData.header = "#ERROR"
+		dialogData.message = reason
+		dialogData.image = $"ui/menu/common/dialog_error"
+
+		#if PC_PROG
```

Comment:
Why is this used here? ^^

Reply:
bump

Reply:
Ugh, I feel like we had a discussion about this somewhere but I don't remember any more what the outcome of this was. If anyone could LMK, please do <3

Reply:
`PC_PROG` is just a check if the engine is running in a PC platform, so it would exclude console executions, probably the formatting of that thing wouldn't matter given Northstar is only for PC.